AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/RemoteSyncReconciler.swift

```diff
@@ -42,18 +42,35 @@ struct RemoteSyncReconciler {
     /// - Returns: disposition of models to apply locally
     static func getDispositions(_ remoteModels: [RemoteModel],
                                 localMetadatas: [LocalMetadata]) -> [Disposition] {
-        guard !remoteModels.isEmpty else {
-            return []
-        }
-
-        guard !localMetadatas.isEmpty else {
-            return remoteModels.compactMap { getDisposition($0, localMetadata: nil) }
+        
+        let remoteModelsGroupedById = remoteModels.reduce(into: [:]) { $0[$1.model.id] = ($0[$1.model.id] ?? []) + [$1] }
```

Comment:
Why we get multiple remoteModels for the same instance? Maybe we should just take remoteModel with highest version for each model instance and ignore the rest?

Reply:
yeah, take the hightest version sounds good to me.

here is the suggest code:
```suggestion
    /// Reconciles the incoming `remoteModels` against the local metadata to get the disposition
    ///
    /// GroupBy the remoteModels by model identifier and apply only the latest version of the remoteModel
    ///
    /// - Parameters:
    ///   - remoteModels: models retrieved from the remote store
    ///   - localMetadatas: metadata retrieved from the local store
    /// - Returns: disposition of models to apply locally
    static func getDispositions(
        _ remoteModels: [RemoteModel],
        localMetadatas: [LocalMetadata]
    ) -> [Disposition] {
        let remoteModelsGroupByIdentifier = remoteModels.reduce([String: [RemoteModel]]()) {
            $0.merging([
                $1.model.identifier: ($0[$1.model.identifier] ?? []) + [$1]
            ], uniquingKeysWith: { $1 })
        }

        let optimizedRemoteModels = remoteModelsGroupByIdentifier.values.compactMap {
            $0.sorted(by: { $0.syncMetadata.version > $1.syncMetadata.version }).first
        }

        guard !optimizedRemoteModels.isEmpty else {
            return []
        }

        guard !localMetadatas.isEmpty else {
            return optimizedRemoteModels.compactMap { getDisposition($0, localMetadata: nil) }
        }

        let metadataBymodelId = localMetadatas.reduce(into: [:]) { $0[$1.modelId] = $1 }
        let dispositions = optimizedRemoteModels.compactMap {
            getDisposition($0, localMetadata: metadataBymodelId[$0.model.identifier])
        }

        return dispositions
    }
```